mungegithub/mungers/approval-handler.go

```diff
@@ -64,15 +61,17 @@ func (*ApprovalHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
 
 // Munge is the workhorse the will actually make updates to the PR
 // The algorithm goes as:
-// - Initially, we set up approverSet
-//   - Go through all comments after latest commit. If any approver said "/approve", add him to approverSet.
-// - For each file, we see if any approver of this file is in approverSet.
+// - Initially, we build two sets, an approverSet and a cancelSet
+//   - Go through all comments after latest commit.
+//	- If anyone said "/approve", add them to approverSet.
+//	- If anyone said "/approve cancel", add them to cancelSet.
+// - Then, for each file, we see if any approver of this file is in approverSet or cancelSet
 //   - An approver of a file is defined as:
-//     - It's known that each dir has a list of approvers. (This might not hold true. For usability, current situation is enough.)
-//     - Approver of a dir is also the approver of child dirs.
-//   - We look at top N (default 3) level dir approvers. For example, for file "/a/b/c/d/e", we might search for approver from
-//     "/", "/a/", "/a/b/"
-// - Iff all files has been approved, the bot will add "approved" label.
+//     - Someone listed as an "approver" in an OWNERS file in the files directory OR
+//     - in one of the file's parent directories
+//   - If a valid approver is in cancelSet, we remove the approve label (if there is one) and return immediately
+// - Iff all files have been approved, the bot will add the "approved" label.
+// - This implies that if any file doesn't have "approved" from a valid approver, the approval label will not be applied
```

Comment:
Not for this PR, but we definitely need to tell people what files are not approved, and who can approve